### PR TITLE
Rewrite ApoExistenceValidator for cocina objects

### LINKED ISSUE #3306: Rewrite APOExistenceValidator to take a cocina item as its input
https://github.com/sul-dlss/dor-services-app/blob/7f552c9b2fbcd93e6bfa1d80f85c2a609f450fa0/app/validators/cocina/apo_existence_validator.rb

### PR DESCRIPTION
**Pull request by aaron-collier**
## Why was this change made?

Fixes #3306 by using CocinaObjectStore to verify the existence of an APO

## How was this change tested?

Update unit tests
Adds a unit test for ApoExistenceValidator that was missing.

### CODE REVIEW

File: spec/requests/create_collection_spec.rb [hunk 1]
```diff
@@ -4,12 +4,23 @@
...
+                                      version: 1,
+                                      administrative: { hasAdminPolicy: 'druid:hy787xj5878', hasAgreement: 'druid:bb033gt0615' }
+                                    })
+  end
   let(:data) { item.to_json }
   let(:druid) { 'druid:gg777gg7777' }
 
   before do
     allow(Dor::SuriService).to receive(:mint_id).and_return(druid)
     allow(Dor).to receive(:find).and_return(apo)
```
Comment (jcoyne) [inline spec/requests/create_collection_spec.rb:22]: Can you remove this one?

File: spec/requests/create_object_spec.rb [hunk 2]
```diff
@@ -4,13 +4,24 @@
...
+                                      administrative: { hasAdminPolicy: 'druid:hy787xj5878', hasAgreement: 'druid:bb033gt0615' }
+                                    })
+  end
   let(:data) { item.to_json }
   let(:druid) { 'druid:gg777gg7777' }
   let(:search_result) { [] }
 
   before do
     allow(Dor::SuriService).to receive(:mint_id).and_return(druid)
     allow(Dor).to receive(:find).and_return(apo)
```
Comment (jcoyne) [inline spec/requests/create_object_spec.rb:23]: Can you remove this one.

File: spec/requests/update_metadata_spec.rb [hunk 3]
```diff
@@ -6,6 +6,7 @@
   before do
     allow(Dor).to receive(:find).with(druid).and_return(item)
     allow(Dor).to receive(:find).with(apo_druid).and_return(apo)
```
Comment (jcoyne) [inline spec/requests/update_metadata_spec.rb:8]: remove?

File: spec/requests/update_metadata_spec.rb [hunk 4]
```diff
@@ -24,6 +25,16 @@
 
   let(:collection) { Dor::Collection.new(pid: 'druid:xx888xx7777') }
   let(:apo) { Dor::AdminPolicyObject.new(pid: apo_druid) }
```
Comment (jcoyne) [inline spec/requests/update_metadata_spec.rb:27]: remove this?

File: spec/requests/create_object_spec.rb [hunk 5]
```diff
@@ -4,13 +4,24 @@
 
 RSpec.describe 'Create object' do
   let(:apo) { Dor::AdminPolicyObject.new(pid: 'druid:dd999df4567') }
```
Comment (jcoyne) [inline spec/requests/create_object_spec.rb:6]: Can this be removed?

File: spec/requests/create_collection_spec.rb [hunk 6]
```diff
@@ -4,12 +4,23 @@
 
 RSpec.describe 'Create object' do
   let(:apo) { Dor::AdminPolicyObject.new(pid: 'druid:dd999df4567') }
```
Comment (jcoyne) [inline spec/requests/create_collection_spec.rb:6]: Can this be removed?

**Review by jcoyne — CHANGES_REQUESTED** [review]

**Comment by aaron-collier** [comment]
@jcoyne I believe I've addressed your comments.

**Review by jcoyne — APPROVED** [review]

**Pull request merged**